### PR TITLE
Cli setup option

### PR DESCRIPTION
**Pull request by idantene**
Adds `meeshkan setup` command (builds directory structure and asks for refresh token)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,7 +1,8 @@
 # Meeshkan client
 
 ## Getting started
-Sign up at [app.meeshkan.com](https://app.meeshkan.com) and get your token. Add the folder `.meeshkan` in your home directory and, inside the folder, add a file named `credentials` with the following format:
+Sign up at [app.meeshkan.com](https://app.meeshkan.com) and get your token.
+You can either run `meeshkan config` to set things up, or manually add the **folder** `.meeshkan` in your home directory and, inside the folder, add a **file** named `credentials` with the following format:
```
Comment (ksaaskil) [inline README.md]: `meeshkan setup` instead of `meeshkan config`?

File: README.md [hunk 2]
```diff
@@ -25,6 +26,7 @@ Commands:
   clear     Clears Meeshkan log and job directories in ~/.meeshkan.
   help      Show this message and exit.
   list      Lists the job queue and status for each job.
+  setup     Configures the Meeshkan client for usage.
```
Comment (ksaaskil) [inline README.md]: "Configures token" or "Configures the client" could be more descriptive/shorter?

File: meeshkan/__main__.py [hunk 3]
```diff
@@ -161,6 +161,20 @@ def help_cmd(ctx):
...
+    """Configures the Meeshkan client for usage."""
+    meeshkan.config.ensure_base_dirs(verbose=False)
+    print("Welcome to Meeshkan!\n")
+    if os.path.isfile(meeshkan.config.CREDENTIALS_FILE):
+        res = input("Credential file already exists! Are you sure you want to overwrite it? [Y]/n: ")
+        if res and res.lower() != "y":  # Any response other than empty or "Y"/"y"
+            print("Aborting")
+            sys.exit(2)
+    token = input("Please enter your refresh token: ")
+    meeshkan.config.Credentials.to_isi(refresh_token=token)
```
Comment (ksaaskil) [inline meeshkan/__main__.py]: Personal stuff like `default report interval` could perhaps be added here at some point as well.

File: meeshkan/core/config.py [hunk 4]
```diff
@@ -29,12 +29,13 @@
...
 
     def create_dir_if_not_exist(path: Path):
         if not path.is_dir():
             # Print instead of logging as loggers may not have been configured yet
-            print("Creating directory {path}".format(path=path))
+            if verbose:
+                print("Creating directory {path}".format(path=path))
             path.mkdir()
 
     create_dir_if_not_exist(BASE_DIR)
```
[the remark this answers is not in the document]
Comment (ksaaskil) [reply meeshkan/core/config.py]: No preference.

File: tests/test_config.py [hunk 5]
```diff
@@ -1,7 +1,22 @@
...
+
+def test_credentials_change_with_error():
+    non_existing_path = Path(tempfile.gettempdir()).joinpath(next(tempfile._get_candidate_names()))
+    with pytest.raises(FileNotFoundError):
+        meeshkan.config.Credentials.to_isi("abc", non_existing_path.joinpath(next(tempfile._get_candidate_names())))
+
+def test_credentials_change_no_error():
+    original_token = meeshkan.config.CREDENTIALS.refresh_token
+    meeshkan.config.Credentials.to_isi("abc", __CREDENTIALS_PATH)
+    assert meeshkan.config.Credentials.from_isi(__CREDENTIALS_PATH).refresh_token == "abc"
```
Comment (ksaaskil) [inline tests/test_config.py]: Does this overwrite `tests/resources/credentials` or?

Comment (ksaaskil) [reply tests/test_config.py]: Ah ok, it's rewritten below. That works, I'd perhaps use a tempfile for such tests myself.

File: README.md (hunk 2 above)
Comment (idantene) [reply README.md]: Also configures the directory structure though

File: meeshkan/__main__.py (hunk 3 above)
Comment (idantene) [reply meeshkan/__main__.py]: Yup. Was also thinking about some other default behavior to include (a la "start service on submit", "stop service with empty queue", etc)

**Pull request merged**